### PR TITLE
MCKIN-9794 - fixed-internal-admin-participant-search

### PR DESCRIPTION
**Pull request by mudassir-hafeez**

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @mudassir-hafeez! I've created [OSPR-3158](https://openedx.atlassian.net/browse/OSPR-3158) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).



We can't start reviewing your pull request until you've submitted a     [signed contributor agreement](https://open.edx.org/wp-content/uploads/2019/01/individual-contributor-agreement.pdf)     or indicated your institutional affiliation. Please see the [CONTRIBUTING](https://github.com/edx/edx-platform/blob/master/CONTRIBUTING.rst)     file for more information.

File: edx_solutions_api_integration/users/views.py [hunk 1]
```diff
@@ -321,6 +351,13 @@ def get_queryset(self):
         courses = css_param_to_list(self.request, 'courses')
         organization_display_name = self.request.query_params.get('organization_display_name', None)
 
+        # filter internal admin course ids
+        internal_courses = self.request.query_params.get('internal_admin_flag', None)
```
Comment (LanghaniKuldeep) [inline edx_solutions_api_integration/users/views.py]: No need to specify None.

File: edx_solutions_api_integration/users/views.py [hunk 2]
```diff
@@ -321,6 +351,13 @@ def get_queryset(self):
...
+        if internal_courses:
+            courses = _get_internal_course_ids(self.request, courses)
+            if not courses:
+                queryset = queryset.none()
```
[the remark this answers is not in the document]
Comment (mudassir-hafeez) [reply edx_solutions_api_integration/users/views.py]: Addressed

File: edx_solutions_api_integration/users/views.py [hunk 3]
```diff
@@ -226,6 +226,32 @@ def _manage_role(course_descriptor, user, role, action):
...
+                courses_filter_list = [
+                    Q(group=existing_group) & Q(course_id__icontains=course) for
+                    course in courses]
+                courses_filter_list = reduce(lambda a, b: a | b, courses_filter_list)
+                members = CourseGroupRelationship.objects.filter(courses_filter_list)
+            else:
+                members = CourseGroupRelationship.objects.filter(group=existing_group)
+            for member in members:
+                internal_course_ids.append(member.course_id)
+    if internal_course_ids:
```
Comment (jawad-khan) [inline edx_solutions_api_integration/users/views.py]: No need for this if.

File: edx_solutions_api_integration/users/views.py [hunk 4]
```diff
@@ -226,6 +226,32 @@ def _manage_role(course_descriptor, user, role, action):
...
+def _get_internal_course_ids(request, courses=[]):
+    """
+    Helper method for filtering/getting course ids of internal admin
+    """
+    internal_course_ids = []
+    group_type = request.query_params.get('type', None)
+    if group_type:
+        groups = Group.objects.select_related("groupprofile").filter(groupprofile__group_type=group_type)
+        for group in groups:
+            existing_group = Group.objects.get(id=group.id)
```
[the remark this answers is not in the document]
Comment (mudassir-hafeez) [reply edx_solutions_api_integration/users/views.py]: Addressed

**Pull request merged**